test/Settings.ts

```diff
@@ -209,185 +123,71 @@ describe('Settings', function () {
 
     describe('Set default', function () {
       it('Resets Node and Matcher URLs', async function () {
-        await setDefaultBtn.click();
-        expect(await nodeUrlInput.getAttribute('value')).to.be.equal(nodeUrl);
-        expect(await matcherUrlInput.getAttribute('value')).to.be.equal(
-          matcherUrl
-        );
+        await NetworkSettingsScreen.setDefaultButton.click();
+        expect(NetworkSettingsScreen.nodeAddress).toHaveValue(nodeUrl);
+        expect(NetworkSettingsScreen.matcherAddress).toHaveValue(matcherUrl);
       });
     });
   });
 
-  describe('Permission control', function () {
+  describe('Permissions control', function () {
     before(async function () {
-      await this.driver
-        .wait(until.elementLocated(By.css('#settingsPermission')), this.wait)
-        .click();
+      await SettingsScreen.permissionsSectionLink.click();
     });
 
     after(async function () {
-      await this.driver.findElement(By.css('div.arrow-back-icon')).click();
+      await TopMenu.backButton.click();
     });
 
     const checkChangingAutoLimitsInResourceSettings = () => {
       describe('Changing auto-limits in resource settings', function () {
         beforeEach(async function () {
-          await this.driver
-            .wait(
-              until.elementLocated(
-                By.xpath(
-                  "//div[contains(@class, 'permissionItem@list')]//button[contains(@class, 'settings@list')]"
-                )
-              ),
-              this.wait
-            )
-            .click();
-          await this.driver.sleep(DEFAULT_ANIMATION_DELAY);
-
-          await this.driver.wait(
-            until.elementIsVisible(
-              this.driver.wait(
-                until.elementLocated(By.css('div#originSettings')),
-                this.wait
-              )
-            ),
-            this.wait
-          );
+          (
+            await PermissionControlSettingsScreen.permissionItems
+          )[0].detailsIcon.click();
+          await browser.pause(DEFAULT_ANIMATION_DELAY);
```

Comment:
I think we have to wait for the modal form here.

Reply:
done